pkg/pillar/zedcloud/send.go

```diff
@@ -614,6 +614,124 @@ func SendOnIntf(ctx *ZedCloudContext, destURL string, intf string, reqlen int64,
 	return nil, nil, senderStatus, errors.New(errStr)
 }
 
+//SendLocal uses local routes to request the data
+func SendLocal(ctx *ZedCloudContext, destURL string, reqlen int64, b *bytes.Buffer) (*http.Response, []byte, error) {
+
+	log := ctx.log
+	var reqURL string
+	var isGet bool
+
+	if strings.HasPrefix(destURL, "http:") {
+		reqURL = destURL
+	} else {
+		if strings.HasPrefix(destURL, "https:") {
+			reqURL = destURL
+		} else {
+			reqURL = "https://" + destURL
+		}
+	}
+
+	if b == nil {
+		isGet = true
+	}
+
+	transport := &http.Transport{
+		TLSClientConfig: ctx.TlsConfig,
+	}
+	// Since we recreate the transport on each call there is no benefit
+	// to keeping the connections open.
+	defer transport.CloseIdleConnections()
+
+	var errorList []error
+	var ipSrc net.IP
+	// Try all addresses
```

Comment:
Maybe we should restrict this to only local ports aka bnX interfaces? That would make it have less opportunity for security issues.

Reply:
Can you please recommend me the way of getting theirs IPs? Probably the function is already here

Reply:
We don't have a function for that. I don't know if @naiming-zededa will be adding one as part of his PR.
It is walking all of the NetworkInstanceStatus and using the BridgeIPAddr field (if it is set; it would be empty for a switch network instance.)

Reply:
Added function `prepareLocalProfileServerMap` which process configuration of network instances to find match with defined localServerURL

Reply:
OK